src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -138,6 +138,11 @@
   ],
 
   properties: [
+    {
```

Comment:
Could this be transient and be calculated by an expression on objId and daoKey?

Reply:
In the case of something like UCJ, the information wouldn't be enough. If that's the case then we would have to make something specific for UCJ where we want the sourceId of the UCJ and also query that to get the object we want as the referenceSummary and call toSummary() on it.

In terms of transient, I  believe it's better to store it because the back-end knows the true intent of what the underlying request is about since the various rules are what instantiate the requests in the first place. If we rely on having the front-end just attempt to calculate what the intent for reference summary is then I think that will restrict the flexibility of referenceSummary.

I'm not opposed to making default behaviour work with objId and daoKey however.

I think saying it's a direct summary of the View Reference doesn't really work with the UCJ cases.

Reply:
I guess thinking of it as a second bit of classification data in addition to  the actual classification property might be clearer.

Reply:
Think if we go down that road, then ComplianceApprovalRequests.referenceSummary will override the expression to call to summary on whatever the sourceId object is.

Reply:
Yeah now I'm thinking and looking over all the rules I had to manually add the change in NP side, the ComplianceApprovalRequests.referenceSummary override will work to handle the UCJ edge case and we could make this transient where the browser can calculate what to show based on grabbing the object with objId and daoKey and calling toSummary on it.